src/coordination/coordinator_instance.cpp

```diff
@@ -353,19 +421,23 @@ auto CoordinatorInstance::UnregisterReplicationInstance(std::string_view instanc
     return UnregisterInstanceCoordinatorStatus::NO_INSTANCE_WITH_NAME;
   }
 
-  auto const is_main = [this](ReplicationInstance const &instance) {
-    return IsMain(instance.InstanceName()) && instance.GetMainUUID() == raft_state_.GetUUID() && instance.IsAlive();
+  auto const is_current_main = [this](ReplicationInstance const &instance) {
+    return raft_state_.IsCurrentMain(instance.InstanceName()) && instance.IsAlive();
   };
 
-  if (is_main(*inst_to_remove)) {
+  if (is_current_main(*inst_to_remove)) {
     return UnregisterInstanceCoordinatorStatus::IS_MAIN;
   }
 
+  if (!raft_state_.AppendOpenLockUnregister(instance_name)) {
```

Comment:
Do we need 10 different opening lock commands, do we distinguish states somehow? Because if not, one command would be much easier and better

Reply:
I agree, my plan was to have it as different, because I didn't know what will I need, this part is being implemented in force sync PR and I will have better state there